apps/studio/src/features/editing-experience/components/LinkEditor/LinkHrefEditor.tsx

```diff
@@ -33,21 +42,40 @@ export const LinkHrefEditor = ({
   description,
   isRequired,
   isInvalid,
+  errorMessage,
+  setErrorMessage,
+  clearErrorMessage,
   pageLinkElement,
   fileLinkElement,
 }: LinkHrefEditorProps) => {
   const linkType = getLinkHrefType(value)
   const [selectedLinkType, setSelectedLinkType] = useState(linkType)
+  const [linkValueHistory, setLinkValueHistory] = useState({
+    ...INITIAL_LINK_VALUE_HISTORY,
+    ...{
+      [linkType]: value,
+    },
+  } as LinkValueHistory)
```

Comment:
not sure why we need the cast here, do you mind explaining it abit more?

Reply:
wasn't familiar with best practise of TS - skill issue, will remove haha